firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -1134,6 +1132,22 @@ class BrowserViewController: UIViewController,
 
     // MARK: - Update content
 
+    func updateContentInHomePanel(_ browserViewType: BrowserViewType) {
+        switch browserViewType {
+        case .normalHomepage:
+            showEmbeddedHomepage(inline: true, isPrivate: false)
+        case .privateHomepage:
+            showEmbeddedHomepage(inline: true, isPrivate: true)
+        case .webview:
+            showEmbeddedWebview()
+            urlBar.locationView.reloadButton.isHidden = false
+        }
+
+        if UIDevice.current.userInterfaceIdiom == .pad {
+            topTabsViewController?.refreshTabs()
+        }
+    }
+
     func updateInContentHomePanel(_ url: URL?, focusUrlBar: Bool = false) {
```

Comment:
I didn't have time to dig into the url use case so left it alone for now to avoid accidentally breaking something. I'll create a task to follow up and dig into if it's actually needed and if it is then to refactor to share code.